psydac/utilities/utils.py

```diff
@@ -5,13 +5,32 @@
 import numpy as np
 
 #===============================================================================
-def refine_array_1d( x, n ):
+def refine_array_1d(x, n, remove_duplicates=True):
+    """Refines a 1D array by create a subdivision between each point
 
-    xr = []
-    for (a,b) in zip(x[:-1],x[1:]):
-        xr.extend( np.linspace( a, b, n, endpoint=False ) )
-    xr.append( x[-1] )
+    Parameters
+    ----------
+    x : ndarray
+        1D array to be refined
+    n : int
+        The number of points in each subdivision
```

Comment:
```suggestion
        Number of subdivisions to be created in each interval (x[i], x[i+1]).
```

Reply:
Done in 8301506

Reply:
Have you?

Reply:
Sorry I completely glossed over this. It's fixed now in [ece414e](https://github.com/pyccel/psydac/pull/193/commits/ece414ed9789c760a0f91550c1da6c44d95d9ae7)